app/domain/authentication/authn_jwt/validate_restrictions_one_to_one.rb

```diff
@@ -0,0 +1,19 @@
+module Authentication
+  module AuthnJwt
+
+    # This class is responsible for retrieving the correct value from the JWT token
+    # of the requested attribute.
+    class ValidateRestrictionsOneToOne
+      def initialize(decoded_token:)
+        @decoded_token = decoded_token
+      end
+
+      def valid_restriction?(restriction)
+        if @decoded_token.key?(restriction.name.to_sym)
+          return @decoded_token[restriction.name.to_sym] == restriction.value
+        end
+        raise Errors::Authentication::AuthnJwt::JwtTokenClaimIsMissing.new(restriction.name)
```

Comment:
I would switch this to first raise an error and then check

```suggestion
        unless @decoded_token.key?(restriction.name.to_sym)
          raise Errors::Authentication::AuthnJwt::JwtTokenClaimIsMissing.new(restriction.name)
        end
        return @decoded_token[restriction.name.to_sym] == restriction.value
```

Reply:
Fixed